src/net.h

```diff
@@ -113,13 +113,16 @@ struct CSerializedNetMsg
     std::string m_type;
 };
 
+/** Different types of connections to a peer. This enum encapsulates the
+ * information we have available at the time of opening or accepting the
+ * connection. Aside from INBOUND, all types are initiated by us. */
 enum class ConnectionType {
-    INBOUND,
-    OUTBOUND,
-    MANUAL,
-    FEELER,
-    BLOCK_RELAY,
-    ADDR_FETCH,
+    INBOUND, /**< peer initiated connections */
+    OUTBOUND, /**< full relay connections (blocks, addrs, txns) made automatically. Addresses selected from AddrMan. */
+    MANUAL, /**< connections to addresses added via addnode or the connect command line argument */
+    FEELER, /**< short lived connections used to test address validity */
+    BLOCK_RELAY, /**< only relay blocks to these automatic outbound connections. Addresses selected from AddrMan. */
+    ADDR_FETCH, /**< short lived connections used to solicit addrs when starting the node without a populated AddrMan */
```

Comment:
I wonder if it might make sense to prefix the different outbound types with OUTBOUND_, so that it's clear to code readers in the future that these are all different types of outbound peers?  In the future, for instance, I'm hoping to propose a way to negotiate block-relay-only connections at connection startup time, and then we'd be able to have both OUTBOUND_BLOCK_RELAY and INBOUND_BLOCK_RELAY peers.

If you were to go this route, then I'd suggest OUTBOUND could be renamed to OUTBOUND_FULL_RELAY or something like that.

Reply:
This was my feedback as well: https://github.com/bitcoin/bitcoin/pull/19316#discussion_r448062841

Reply:
Oops, sorry for the noise and missing the earlier discussion. This would be fine to do later as well once we get around to adding more inbound types.

Reply:
> Oops, sorry for the noise and missing the earlier discussion. 

no worries. would rather you read the code :) 

>  I'm hoping to propose a way to negotiate block-relay-only connections at connection startup time, and then we'd be able to have both OUTBOUND_BLOCK_RELAY and INBOUND_BLOCK_RELAY peers.

oh interesting! curious to see how it develops.
 
I've tended towards sticking with the simpler names because (1) its hard to discern exactly what properties will be relevant for the future eg. ariard [mentioned](https://github.com/bitcoin/bitcoin/pull/19316#pullrequestreview-442115002) manual block relay connections / could lead to maintenance burden as sipa mentioned. (2) it should be easy to rename in the future with a scripted diff (3) just as a personal preference, eg. `OUTBOUND_AUTO_FEELER` feels unnecessarily verbose when we don't have an idea of any other types of feeler connections. or `INBOUND_AUTO_FULL` could be slightly misleading because maybe its actually only blocks being relayed on this connection.